node/flatpak-node-generator.py

```diff
@@ -488,6 +488,39 @@ class Package(NamedTuple):
     lockfile: Path
 
 
+class NodeHeaders:
+    def __init__(self, target: str, runtime: str = None, disturl: str = None):
+        self.target = target
+        if runtime is not None:
+            self.runtime = runtime
+        else:
+            self.runtime = 'node'
+        if disturl is not None:
+            self.disturl = disturl
+        elif self.runtime == 'node':
+            self.disturl = 'http://nodejs.org/dist'
+        elif self.runtime == 'electron':
+            self.disturl = 'https://www.electronjs.org/headers'
+        else:
+            raise ValueError(f'Can\'t guess `disturl` for {self.name} version {self.version}')
+
+    @property
+    def name(self):
+        return f'{self.runtime}-headers'
+
+    @property
+    def version(self):
+        return self.target
+
+    def get_url(self) -> str:
```

Comment:
Nit: most `get_x(self)` functions would be better off as properties (also why the separate `def version()` property if it's identical to `target`? :thinking: )

Reply:
That's for `GeneratorProgress()._format_package()` which expects package object with `name` and `version` attributes.
Hmm, maybe it should use `__str__()` representation of a package object instead?